### PR TITLE
Close #9145: Catch IOException where clearing thumbnails

### LINKED ISSUE #9145: Catch IOException when deleting thumbnail storage
Sometimes when deleting the thumbnails directory with `DiskLruCache.delete` the directory can already have been removed.  There isn't anything else we can do here, so let's just catch the error then.

Ref: https://github.com/mozilla-mobile/fenix/issues/13209

### PR DESCRIPTION
**Pull request by jonalmeida**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/browser/thumbnails/src/main/java/mozilla/components/browser/thumbnails/utils/ThumbnailDiskCache.kt [hunk 1]
```diff
@@ -22,13 +23,19 @@ private const val WEBP_QUALITY = 90
...
         synchronized(thumbnailCacheWriteLock) {
-            getThumbnailCache(context).delete()
-            thumbnailCache = null
+            try {
+                getThumbnailCache(context).delete()
+            } catch (e: IOException) {
+                logger.warn("Thumbnail cache could not be cleared. Perhaps there are none?")
+            } finally {
+                thumbnailCache = null
+            }
```
Comment (pocmo) [inline components/browser/thumbnails/src/main/java/mozilla/components/browser/thumbnails/utils/ThumbnailDiskCache.kt:38]: nit: Technically you do not need to put this into a `finally` block here :)

File: components/browser/thumbnails/src/test/java/mozilla/components/browser/thumbnails/utils/ThumbnailDiskCacheTest.kt [hunk 2]
```diff
@@ -76,4 +79,17 @@ class ThumbnailDiskCacheTest {
...
         assertNull(data)
     }
+
+    @Test
+    fun `Clearing bitmap from disk catch IOException`() {
+        val cache = ThumbnailDiskCache()
+        val lruCache: DiskLruCache = mock()
+        cache.thumbnailCache = lruCache
+
+        `when`(lruCache.delete()).thenAnswer { throw IOException("test") }
```
Comment (pocmo) [inline components/browser/thumbnails/src/test/java/mozilla/components/browser/thumbnails/utils/ThumbnailDiskCacheTest.kt:89]: nit: There's also `thenThrow()`.

**Review by pocmo — APPROVED** [review]

File: components/browser/thumbnails/src/test/java/mozilla/components/browser/thumbnails/utils/ThumbnailDiskCacheTest.kt (hunk 2 above)
Comment (jonalmeida) [reply components/browser/thumbnails/src/test/java/mozilla/components/browser/thumbnails/utils/ThumbnailDiskCacheTest.kt:89]: Oops, I was trying out something with this test and forgo to revert it. :)

File: components/browser/thumbnails/src/main/java/mozilla/components/browser/thumbnails/utils/ThumbnailDiskCache.kt (hunk 1 above)
Comment (jonalmeida) [reply components/browser/thumbnails/src/main/java/mozilla/components/browser/thumbnails/utils/ThumbnailDiskCache.kt:38]: Thanks!

I was curious, so I looked at the generated Java code. This is what it looks like with the `finally` block:
```java
synchronized(var2){}

try {
   boolean var5 = false;

   try {
      this.getThumbnailCache(context).delete();
   } catch (IOException var13) {
      Logger.warn$default(this.logger, "Thumbnail cache could not be cleared. Perhaps there are none?", (Throwable)null, 2, (Object)null);
   } finally {
      this.thumbnailCache = (DiskLruCache)null;
   }

   Unit var16 = Unit.INSTANCE;
} finally {
   ;
}
```

And here it is without the `finally`:
```java
synchronized(var2) {
   boolean var5 = false;

   try {
      this.getThumbnailCache(context).delete();
   } catch (IOException var7) {
      Logger.warn$default(this.logger, "Thumbnail cache could not be cleared. Perhaps there are none?", (Throwable)null, 2, (Object)null);
   }

   this.thumbnailCache = (DiskLruCache)null;
   Unit var9 = Unit.INSTANCE;
}

```

Questions for self:
 - Why does the `finally` block stop the try block from being executed in the `synchronized` block?
 - The nested `finally` is also very confusing and seems super redundant. 🤔

**Pull request merged**